kpops/pipeline_generator/pipeline.py

```diff
@@ -67,6 +77,55 @@ def validate_unique_names(self) -> None:
                 f"step names should be unique. duplicate step names: {', '.join(duplicates)}"
             )
 
+    def generate_graph(self) -> None:
+        for component in self.components:
+            all_input_topics = self.__get_all_input_topics(component)
+            all_output_topics = self.__get_all_output_topics(component)
+
+            component_node_name = self.__get_vertex_component_name(component)
+            self.graph.add_node(component_node_name)
+
+            self.__add_ingoing_edges(all_input_topics, component_node_name)
+            self.__add_outgoing_edges(all_output_topics, component_node_name)
+
+    def __add_outgoing_edges(
+        self, all_output_topics: list[str], component_node_name: str
+    ) -> None:
+        for output_topic in all_output_topics:
+            self.graph.add_node(output_topic)
+            self.graph.add_edge(component_node_name, output_topic)
+
+    def __add_ingoing_edges(
+        self, all_input_topics: list[str], component_node_name: str
+    ) -> None:
+        for input_topic in all_input_topics:
+            self.graph.add_node(input_topic)
+            self.graph.add_edge(input_topic, component_node_name)
+
+    @staticmethod
+    def __get_vertex_component_name(component: PipelineComponent) -> str:
+        return f"component-{component.name}"
+
+    def __get_all_output_topics(self, component: PipelineComponent) -> list[str]:
+        all_output_topics: list[str] = []
+        output_topic = component.get_output_topic()
+        extra_output_topics = component.get_extra_output_topics()
+        if output_topic is not None:
+            all_output_topics.append(output_topic)
+        if extra_output_topics:
+            all_output_topics.extend(list(extra_output_topics.values()))
+        return all_output_topics
+
+    def __get_all_input_topics(self, component: PipelineComponent) -> list[str]:
```

Comment:
make these two functions properties `inputs` and `outputs` inside `PipelineComponent` and return an `Iterator[str]`. no need to collect everything in a list
this could just be 
```python
yield from self.get_input_topics()
...
```